src/west/commands.py

```diff
@@ -246,15 +246,32 @@ def git_version_info(self):
         '''
         if not hasattr(self, '_git_ver'):
             self.die_if_no_git()
-            # raw_ver usually looks like '2.31.1'
-            raw_ver = self.check_output(
-                [self._git, '--version']).decode().strip().split()[-1]
-            # match just the numeric prefix, in case we get something like
-            # '2.31.1-EXTRAVERSION' in some situations. The major/minor/patch
-            # group names are purely for clarity.
-            match = re.match(
-                r'^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)', raw_ver)
-            self._git_ver = tuple(int(group) for group in match.groups())
+            # The raw 'git --version' output varies by platform and due to
+            # i18n. Examples in English are:
+            #
+            # git version 2.25.1
+            # git version 2.28.0.windows.1
+            # git version 2.24.3 (Apple Git-128)
+            #
+            # We handle this by matching the first string in the output
+            # that has a prefix that looks like a semver. Hopefully this
+            # is general enough.
+
+            raw_ver = self.check_output([self._git, '--version']).decode()
+
+            # Match just the numeric prefix, to handle '.windows.1'
+            # type suffixes.
+            ver_re = re.compile(r'^(\d+)\.(\d+)\.(\d+)')
+
+            for word in raw_ver.strip().split():
```

Comment:
Have you considered `\b`oundary as in `r'\b(\d+)\.(\d+)\.(\d+)'` instead of the `.split()`? It's not equivalent as it would match `(2.8.1)` for instance but it would make the code much shorter.

Reply:
> Have you considered `\b`oundary as in `r'\b(\d+)\.(\d+)\.(\d+)'` instead of the `.split()`? It's not equivalent as it would match `(2.8.1)` for instance but it would make the code much shorter.

I've golfed it down a few lines.